### PR TITLE
Feature/amp

### PR DESCRIPTION
**Pull request by BloodAxe**
## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest .` ? 
- [ ] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?

- added fast_zero_grad support to AMPOptimizerCallback
- scheduler callback now gets learning rate using get_last_lr() method from the scheduler object (As PyTorch suggests)
- added supported `find_unused_parameters` option to use with DDP to detect whether there are any unused parameters of the model.

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

PS
- [x] I know, that I could [join slack](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for pull request discussion.

### CODE REVIEW

File: catalyst/callbacks/scheduler.py [hunk 1]
```diff
@@ -115,9 +115,12 @@ def _scheduler_step(
     ):
         if isinstance(scheduler, torch.optim.lr_scheduler.ReduceLROnPlateau):
             scheduler.step(reduced_metric)
-            lr = scheduler.optimizer.param_groups[0]["lr"]
```
Comment (and-kul) [inline catalyst/callbacks/scheduler.py:118]: Hi @BloodAxe 
Unfortunately, `torch.optim.lr_scheduler.ReduceLROnPlateau` doesn't inherit from `_LRScheduler`, so we cannot use neither `get_last_lr()` nor `get_lr()` for it

**Review by bagxi — APPROVED** [review]

File: catalyst/callbacks/optimizer.py [hunk 2]
```diff
@@ -370,8 +372,10 @@ def on_batch_end(self, runner: "IRunner") -> None:
             self.grad_step(
                 optimizer=self._optimizer, grad_clip_fn=self.grad_clip_fn,
             )
-
-            maybe_recursive_call(self._optimizer, "zero_grad")
+            if not self.use_fast_zero_grad:
+                maybe_recursive_call(self._optimizer, "zero_grad")
+            else:
+                maybe_recursive_call(self._optimizer, zero_grad)
```
Comment (Scitator) [inline catalyst/callbacks/optimizer.py:378]: what's the difference? 🤔

**Comment by Scitator** [comment]
@BloodAxe could you please merge with master and check the codestyle? I think that could be enough ;)

**Comment by Scitator** [comment]
already merged with https://github.com/catalyst-team/catalyst/pull/1007
